Sprint-2/extend/format-time.js

```diff
@@ -1,24 +1,94 @@
 // This is the latest solution to the problem from the prep.
+// https://programming.codeyourfuture.io/structuring-data/sprints/2/prep/
 // Your task is to write tests for as many different groups of input data or edge cases as you can, and fix any bugs you find.
 
 function formatAs12HourClock(time) {
+  // check my time is valid first
+  validateTimeInput(time);
   const hours = Number(time.slice(0, 2));
+  // handle midnight
+  if (hours === 0) return "12:00 am";
```

Comment:
Woah there, why did all the minutes become `00`?

Could a few tests with different hour and minute values have helped to detect this bug? :)

Reply:
You're meant to build up the tests in the next sprint is the issue really.

Reply:
This bug specifically was what made me create this exercise 😂  I don't believe the 12-hour-clock exercise comes back in sprint 3?